example/App.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react'
+import {
+  SafeAreaView,
+  StyleSheet,
+  FlatList,
+  ActivityIndicator,
+  TouchableOpacity,
+  Text,
+  View
+} from 'react-native'
+
+import { BLEmanager } from './index'
+import PrimaryButton from './src/components/PrimaryButton'
+
+const App = () => {
+  const [isLoading, setIsLoading] = React.useState(false)
+  const [isScanning, setIsScanning] = React.useState(false)
+  const [devices, setDevices] = React.useState([])
+
+  React.useEffect(() => {
+    const subscription = BLEmanager.onStateChange((state) => {
+      console.log('BLE stack status: ', state)
+    }, true);
+    return () => subscription.remove();
+  }, [BLEmanager]);
+
+  const handleStartDeviceScan = () => {
+    setIsScanning(true)
+    handleStartLoading()
+    console.log('Scanning started!')
+
+    BLEmanager.startDeviceScan(null, null, (error, device) => {
+      if (error) {
+        console.log('Error! Scanning devices: ', error)
+        return
+      }
+
+      setDevices(prevState => {
+        const duplicat = prevState.find(item => item.id === device.id)
+        if (!duplicat) {
+          return [...prevState, {...device, isConnected: false}]
+        } else {
+          return [...prevState]
+        }
+      })
+    })
+    handleStopLoading()
+  }
+
+  const handleStopDeviceScan = () => {
+    setIsScanning(false)
+    handleStartLoading()
+    BLEmanager.stopDeviceScan();
+    handleStopLoading()
+    console.log('Scanning stopped!')
+  }
+
+  const handleConnectToDevice = async (deviceId) => {
+    handleStartLoading()
+    try {
+      const connectedDevice = await BLEmanager.connectToDevice(deviceId)
+      console.log('Connected to device: ', connectedDevice)
+
+      const connectedDeviceIndex = devices.findIndex(device => device.id === connectedDevice.id)
+      const devicesArr = devices
+      devicesArr[connectedDeviceIndex] = { ...devicesArr[connectedDeviceIndex], isConnected: true }
+      setDevices(devicesArr)
+
+      const characteristics = await connectedDevice.discoverAllServicesAndCharacteristics()
+      console.log('Device services and characteristics: ', characteristics) 
+    } catch (error) {
+      console.log('Error! Connecting to device: ', error)
```

Comment:
We should communicate the errors to the user somehow, not just through the console